catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorHandler.java

```diff
@@ -373,6 +374,10 @@ public static Object getSensorValue(Sensors sensor) {
 				return listeningLanguageSensor;
 			case USER_LANGUAGE:
 				return userLocaleTag;
+			case STAGE_WIDTH:
+				return (double) ScreenValues.SCREEN_WIDTH;
+			case STAGE_HEIGHT:
+				return (double) ScreenValues.SCREEN_HEIGHT;
```

Comment:
```suggestion
			case STAGE_WIDTH:
				return (double) getCurrentProject().getXmlHeader().virtualScreenWidth;
			case STAGE_HEIGHT:
				return (double) getCurrentProject().getXmlHeader().virtualScreenHeight;
```
Also needed are the correct imports for these.

ScreenValues.SCREEN_... are not always the correct values: they only are the **_current_** stage's size in pixels on the screen of the physical device. However, we want the **_stage's_** virtual width and height, from the time when the project was created. These values are different, for instance when downloading projects from other users (the project from the original phone has a different resolution than the phone to which it was downloaded), or when changing the resolution on one's phone in the Android settings in case the phone supports this, or when opening Catroid in a pop-up or split-screen size canvas in case the phone or Android version supports this. 

Note that the testStageWidthStageHeightSensor.catrobat should only test for the precise values in the header of the code.xml file of the project, see screenshots below. 
![image](https://user-images.githubusercontent.com/1747888/162814025-68ca63ac-f9bf-4a1e-a2d1-708626e6c9a9.png)

Delete all other asserts as they are not necessary. Here's the whole CLT that does what is expected of such a test (please also delete the unnecessary variables):
![image](https://user-images.githubusercontent.com/1747888/162814725-ad70b26d-1442-44a1-8a0a-e998177ea43c.png)